recipes/java-jdk/meta.yaml

```diff
@@ -0,0 +1,31 @@
+{% set version = "8.45.14" %}
+
+package:
+  name: java-jdk
+  version: {{ version }}
+
+# Cannot use source section because of cookies blocking download
+#source:
+#  fn: jre-8u25-linux-x64.tar.gz
+#  url: http://download.oracle.com/otn-pub/java/jdk/8u25-b17/jre-8u25-linux-x64.tar.gz
+
+build:
+  number: 0
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - curl  # [win]
+
+test:
+  commands:
+    - java -help
+
+about:
+  home: http://www.java.com/en/
+  license: "Oracle Binary Code License Agreement for the Java SE Platform Products and JavaFX"
```

Comment:
[From Oracle's website](http://www.oracle.com/technetwork/java/javase/terms/license/index.html), here is the relevant part of the license:

> C. LICENSE TO DISTRIBUTE SOFTWARE. Subject to the terms and conditions of this Agreement and restrictions and exceptions set forth in the README File, including, but not limited to the Java Technology Restrictions and Limitations on Redistribution of these Supplemental Terms, Oracle grants you a non-exclusive, non-transferable, limited license without fees to reproduce and distribute the Software, provided that (i) you distribute the Software complete and unmodified and only bundled as part of, and for the sole purpose of running, your Programs, (ii) the Programs add significant and primary functionality to the Software, (iii) you do not distribute additional software intended to replace any component(s) of the Software, (iv) you do not remove or alter any proprietary legends or notices contained in the Software, (v) you only distribute the Software subject to a license agreement that: (a) is a complete, unmodified reproduction of this Agreement; or (b) protects Oracle's interests consistent with the terms contained in this Agreement and that includes the notice set forth in Section H, and (vi) you agree to defend and indemnify Oracle and its licensors from and against any damages, costs, liabilities, settlement amounts and/or expenses (including attorneys' fees) incurred in connection with any claim, lawsuit or action by any third party that arises or results from the use or distribution of any and all Programs and/or Software. The license set forth in this Section C does not extend to the Software identified in Section G.

and 

> G. LIMITATIONS ON REDISTRIBUTION. You may not redistribute or otherwise transfer patches, bug fixes or updates made available by Oracle through Oracle Premier Support, including those made available under Oracle's Java SE Support program.

I think we are within the license restrictions here. We are not patching the source code, the sole purpose is to run programs (conda and other conda packages), we are not modifying the legends or notices in the software, we have a complete and unmodified copy of this license, and we don't fall into the exemptions for the premiere version. 

[The README](http://www.oracle.com/technetwork/java/javase/jdk-8-readme-2095712.html) has more clauses about which files may be included that probably bears further digging in to.